calcom/cal.com
# fix: meeting ended trigger for webhooks and zapier sometimes not working

## What does this PR do?

This PR addresses the issue of the Zapier and workflow meeting ended trigger not working properly. To fix this issue, we replaced the scheduling approach from using `node-schedule` with a cron instead.

The reason we cannot use `node-schedule` for this use case is that this library is only intended for in-process scheduling. Here is an excerpt from their [documentation](https://www.npmjs.com/package/node-schedule):

> Note that Node Schedule is designed for in-process scheduling, i.e. scheduled jobs will only fire as long as your script is running, and the schedule will disappear when execution completes. If you need to schedule jobs that will persist even when your script isn't running, consider using actual [cron](http://unixhelp.ed.ac.uk/CGI/man-cgi?crontab+5).

Fixes: https://github.com/calcom/cal.com/issues/9347

## Type of change




## How should this be tested?

1. Set up Zapier and a webhook with a meeting ended trigger.
2. Schedule multiple meetings at different times (recommended: add a description with "should be triggered at xx:xx").
3. Wait and see if the triggers are executed at the correct time.
4. Check the database to ensure that finished jobs are cleaned correctly.

## Mandatory Tasks


## Checklist



- I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code does follow the style guidelines of this project
- I commented my code, particularly in hard-to-understand areas
- I have checked if my PR needs changes to the documentation
- I have checked if my changes generate no new warnings
- I have checked if new and existing unit tests pass locally with my changes
